src/main/java/Sakura/TaskList.java

```diff
@@ -0,0 +1,154 @@
+package Sakura;
+import java.time.format.DateTimeParseException;
+import java.util.Arrays;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Represents a list for all tasks similar to array lists.
+ */
+public class TaskList {
+    public final ArrayList<Task> tasks = new ArrayList<>();
+
+    /**
+     * Constructor for TaskList.
+     *
+     * @param tasks tasks varargs of tasks to be added to the list
+     */
+    public TaskList(Task... tasks) {
+        this.tasks.addAll(Arrays.asList(tasks));
+    }
+
+    /**
+     * Obtain the number of items in the list.
+     *
+     * @return the length of the list of tasks.
+     */
+    public int getLength() {
+        return this.tasks.size();
+    }
+
+    /**
+     * Obtain the task in the list at the specific index.
+     *
+     * @param taskIndex the index of the target task.
+     * @return the target task found at the specific index.
+     */
+    public Task get(int taskIndex) {
+        return this.tasks.get(taskIndex - 1);
+    }
+
+    /**
+     * Adds a task to the task list via the addDescrpition method.
+     *
+     * @param input string command given by user.
+     */
+    public void addTask(String input) {
+        try {
+            if (input.toLowerCase().startsWith("todo")) {
+                if (input.split(" ").length == 1) {
+                    SakuraException.invalidTodo();
+                } else {
+                    Todo todo = new Todo(input.substring("todo ".length()));

```

Comment:
I like how `"todo ".length()` was not replaced with the magic number `5`!

This helps me understand that we are getting the name of the task by omitting the first five characters.

I should adopt this in my code as well.